modules/backup-gcs/client.go

```diff
@@ -324,20 +290,20 @@ func (g *gcsClient) Write(ctx context.Context, backupID, key string, r io.ReadCl
 	return written, nil
 }
 
-func (g *gcsClient) Read(ctx context.Context, backupID, key string, w io.WriteCloser) (int64, error) {
+func (g *gcsClient) Read(ctx context.Context, backupID, key, bucketName, bucketPath string, w io.WriteCloser) (int64, error) {
 	defer w.Close()
 
-	bucket, err := g.findBucket(ctx)
+	bucket, err := g.findBucket(ctx, bucketName)
 	if err != nil {
-		err = fmt.Errorf("find bucket: %w", err)
+		err = fmt.Errorf("(read) find bucket: '%w'", err)
```

Comment:
can we remove (read)? and also remove single quotes from '%w'

Reply:
can we have unique error messages?  and the single quotes are needed to show empty errors

Reply:
and as above, find bucket should be removed